staging/src/k8s.io/apiserver/pkg/authentication/authenticator/metrics.go

```diff
@@ -75,21 +76,43 @@ func init() {
 	legacyregistry.MustRegister(authenticationLatency)
 }
 
-func recordAuthMetrics(resp *authenticator.Response, ok bool, err error, apiAudiences authenticator.Audiences, authStart time.Time) {
-	var resultLabel string
+// ResetMetrics resets all auth-related metrics - intended for use in unit tests only.
+// Since prometheus' gatherer is global, other tests may have updated metrics already, so
```

Comment:
😞

Reply:
One of our future initiatives is to remove the use of the global metrics registry. You can already use a registry that you pass through your code and register metrics onto it. For now you would just pass the global registry but for tests you could already use a local one.

Reply:
You can create a test registry by [NewFakeKubeRegistry](https://github.com/kubernetes/kubernetes/blob/5708511499fe500ae3b4bbd40204cef382f652e8/staging/src/k8s.io/component-base/metrics/testutil/testutil.go#L57). If it can't fully fit your case, please feel free let me know.

Reply:
Thank for the tip on NewFakeKubeRegistry.
Using an isolated fake now.